plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/AbstractPackageJSONValidatorExtension.java

```diff
@@ -284,7 +286,11 @@ protected boolean isResponsible(Map<Object, Object> context, EObject eObject) {
 		if (!fileName.equals(IN4JSProject.PACKAGE_JSON)) {
 			return false;
 		}
-		IN4JSProject project = n4jsCore.findProject(pckjsonUri).get();
+		Optional<? extends IN4JSProject> optProject = n4jsCore.findProject(pckjsonUri);
+		if (!optProject.isPresent()) {
+			return false;
```

Comment:
The exception thrown at this point before your fix hinted us towards all the other issues. So logging an error would be good, here.

Reply:
True, but you could already add something like `LOGGER.error(...)` to make sure we won't forget this at this location.

Reply:
or we add a general assertion for the redirected output string

Reply:
I don't understand this idea.
What I often do to detect race conditions: add a lot of logging and run code in a loop. If one scenario fails, compare the logs of good and bad cases. Would be great, if that would work for the AbstractIdeTest.

Reply:
We talk about `AbstractIdeTest#redirectPrintStreams()` right? For testing/debugging it might be an idea to comment out the line inside that method body. As a result, the redirection would be disabled. Regarding "general assertion" I meant that we can still get the output via `SystemOutRedirecter#getSystemOut()` as a default and check that for certain things like `!getSystemOut().contains("Exception")` (If that helps?).